pallets/avn/src/lib.rs

```diff
@@ -398,13 +407,15 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
         trace!("Avn pallet new session entrypoint");
         // Update the list of validators if it has changed
         let mut disabled_avn_validators: Vec<T::AccountId> = vec![];
-        let mut active_avn_validators: Vec<Validator<T::AuthorityId, T::AccountId>> = vec![];
+        let mut active_avn_validators = WeakBoundedVec::default();
 
         validators.for_each(|x| {
             if T::DisabledValidatorChecker::is_disabled(x.0) {
                 disabled_avn_validators.push(x.0.clone());
             } else {
-                active_avn_validators.push(Validator::new(x.0.clone(), x.1));
+                if let Err(_) = active_avn_validators.try_push(Validator::new(x.0.clone(), x.1)) {
+                    return // breaking out of the for_each
+                }
```

Comment:
I think we should make this an infallible path. `active_avn_validators` can remain a normal vector and then on lone 423, we create a WeakBoundedVec with force_from and put it into storage.